Mathlib/Algebra/Ring/Semireal/Defs.lean

```diff
@@ -28,19 +26,22 @@ semireal.
 variable (R : Type*)
 
 /--
-A semireal ring is a non-trivial commutative ring (with unit) in which `-1` is *not* a sum of
-squares. Note that `-1` does not make sense in a semiring. Below we define the class `IsSemireal R`
-for all additive monoid `R` equipped with a multiplication, a multiplicative unit and a negation.
+A semireal ring is a commutative ring (with unit) in which `-1` is *not* a sum of
+squares. We define the class `IsSemireal R` for all structures equipped with
+a multiplication, an addition, a multiplicative unit and an additive unit.
 -/
 @[mk_iff]
-class IsSemireal [AddMonoid R] [Mul R] [One R] [Neg R] : Prop where
-  non_trivial         : (0 : R) ≠ 1
-  not_isSumSq_neg_one : ¬IsSumSq (-1 : R)
+class IsSemireal [Add R] [Mul R] [One R] [Zero R] : Prop where
+  add_one_ne_zero_of_isSumSq {a : R} (ssa : IsSumSq a) : 1 + a ≠ 0
 
 @[deprecated (since := "2024-08-09")] alias isSemireal := IsSemireal
 @[deprecated (since := "2024-08-09")] alias isSemireal.neg_one_not_SumSq :=
-  IsSemireal.not_isSumSq_neg_one
+  IsSemireal.add_one_ne_zero_of_isSumSq
 
-instance [LinearOrderedRing R] : IsSemireal R where
-  non_trivial := zero_ne_one
-  not_isSumSq_neg_one := fun h ↦ (not_le (α := R)).2 neg_one_lt_zero h.nonneg
+theorem IsSemireal.not_isSumSq_neg_one {R : Type*} [AddGroup R] [One R] [Mul R] [IsSemireal R]:
+    ¬ IsSumSq (-1 : R) := (by simpa using add_one_ne_zero_of_isSumSq ·)
```

Comment:
Doesn't this work?
```suggestion
    ¬ IsSumSq (-1 : R) := by simpa using add_one_ne_zero_of_isSumSq
```

Reply:
Typeclass inference on `Semireal ?_` fails